### PR TITLE
Add live outputs from the loop tail to the loop params

### PR DESCRIPTION
**Pull request by ezrosent**
Added a test that goes from failing to passing with this change. In this case the `ans` variable is overwritten in the loop body but it is still used outside so it needs to be given an input and output.

Note that for variables that are truly unbound at this point we may have to add a placeholder variable to get them working.

### CODE REVIEW

File: tests/passing/small/sqrt.bril [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# ARGS: 16
```
Comment (yihozhang) [inline tests/passing/small/sqrt.bril:1]: nit: sqrt might be a misleading name since the simplified example isn't doing sqrt

**Review by yihozhang — APPROVED** [review]
The fix looks good but gramschmidt.bril is still failing due to non-linearity. Could be related to extraction.

**Review by oflatt — APPROVED** [review]
Great!

File: tests/passing/small/sqrt.bril (hunk 1 above)
Comment (ezrosent) [reply tests/passing/small/sqrt.bril]: Went ahead and made the test more real

**Pull request merged**